test/integration/device-api/v2.spec.ts

```diff
@@ -0,0 +1,108 @@
+import { expect } from 'chai';
+import * as express from 'express';
+import { SinonStub, stub } from 'sinon';
+import * as request from 'supertest';
+
+import * as config from '~/src/config';
+import * as db from '~/src/db';
+import * as deviceApi from '~/src/device-api';
+import * as actions from '~/src/device-api/actions';
+import * as v2 from '~/src/device-api/v2';
+import { UpdatesLockedError } from '~/lib/errors';
+
+// All routes that require Authorization are integration tests due to
+// the api-key module relying on the database.
+describe('device-api/v2', () => {
+	let api: express.Application;
+
+	before(async () => {
+		await config.initialized();
+
+		// `api` is a private property on SupervisorAPI but
+		// passing it directly to supertest is easier than
+		// setting up an API listen port & timeout
+		api = new deviceApi.SupervisorAPI({
+			routers: [v2.router],
+			healthchecks: [],
+			// @ts-expect-error
+		}).api;
+	});
+
+	describe('POST /v2/applications/:appId/restart', () => {
```

Comment:
Same with tests here, anything is already tested within the lifecycle tests should be removed here. I know right now you are only testing v1 tests within lifecycle, but it may be useful when you are doing the v2 tests

Reply:
Are the input validation tests the ones you're referring to? I don't test any input validation in the lifecycle tests (actions.spec.ts). As in https://github.com/balena-os/balena-supervisor/pull/2048#discussion_r1065223879, should I remove this for now since I plan to deduplicate with an input validation middleware feature after the API test migration?

Reply:
As I said above, I agree you are not testing everything on the lifecycle tests so duplication right now is relatively minimal. Since you have a plan to improve I think you can keep it this way